### PR TITLE
Initial proposal for the Core TCK

### PR DESCRIPTION
**Pull request by cescoffier**
This PR

* updates the current code base to Reactive Streams Operators 1.0
* Fix javadoc
* define the core TCK (signature, scope...)

### CODE REVIEW

File: core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ServiceLoader.load(ArchiveExtender.class).iterator().forEachRemaining(ext -> ext.extend(archive));
+    return archive;
+  }
+
+  @Inject
+  private SimpleIncomingBean simple;
+
+  @Test
+  public void testReceptionWithValues() {
+    assertThat(simple.getValues()).containsExactlyElementsOf(StringSource.VALUES);
```
Comment (jroper) [inline core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java]: Is there a race condition here? How do we know that the bean has finished consuming all the values when the test runs?

File: core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/scope/ApplicationScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      .addClasses(ApplicationBeans.class);
+  }
+
+  @Inject
+  private ApplicationBeans bean;
+
+  @Test
+  public void verify() {
+    assertThat(bean.getList()).containsExactly(1);
+    assertThat(ApplicationBeans.getStaticList()).containsExactly(1);
```
Comment (jroper) [inline core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/scope/ApplicationScopeTest.java]: Likewise, is there a race condition here?

File: core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java (hunk 1 above)
Comment (cescoffier) [reply core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java]: Would be better with `await`.

Comment (cescoffier) [reply core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java]: Let's use `await` everywhere.

Comment (cescoffier) [reply core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/SimpleIncomingTest.java]: Fixed

File: core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/scope/ApplicationScopeTest.java (hunk 2 above)
Comment (cescoffier) [reply core-tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/scope/ApplicationScopeTest.java]: Fixed.

**Comment by cescoffier** [comment]
@jroper I've updated the PR to add awaitility and the related statements to avoid the race conditions.. How should we proceed?

**Pull request merged**